Yoast/yoast-components
# Remove svg files, the Icon component, and svg-react-loader

## Summary

This PR can be summarized in the following changelog entry:

* REmoved the Icon component, replaced by the new SvgIcon React component

## Relevant technical choices:

- this PR is branched out from `stories/415-styled-styled-section`
- removes the Icon component
- removes the svg files from `style-guide/svg`
- uses StyledSection where possible
- adapt other components, e.g. IconButton, IconButtonToggle, to the new SvgIcon component introduced in `stories/415-styled-styled-section`
- removes svg-react-loader

I've looked for all the imports of icons in the other repos and tried to address them. Please help me check no other usages of the SVG files and hte `<Icon>` component have survived and nothing breaks 🙂 
@Xyfi please check the `svg-react-loader` stuff, I have no idea what I've done there 😆 

## Test instructions

- check all the components with icons work correctly in the standalone version
- switch both free and premium to their `stories/415-styled-styled-section` branches if they're not merged yet
- yarn link this branch to free and premium, build the assets, and check all the icons
- on Premium, you will get an error when building the CSS because `StyledSection.scss` was removed, this is covered on free in https://github.com/Yoast/wordpress-seo/pull/8812 I guess your best option is to temporarily remove that import manually

Fixes #420
